sw/host/opentitanlib/src/transport/hyperdebug/servo_micro.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use std::rc::Rc;
+
+use crate::io::gpio::{GpioPin, PinMode, PullMode};
+use crate::transport::hyperdebug::{Flavor, Inner, StandardFlavor, VID_GOOGLE};
+use crate::transport::{TransportError, TransportInterfaceType};
+
+/// The Servo Micro is used to bring up GSC and EC chips sitting inside a computing device, such
+/// that those GSC chips can provide Case Closed Debugging support to allow bringup of the rest of
+/// the computing device. Servo devices happen to speak almost exactly the same USB protocol as
+/// HyperDebug. This `Flavor` implementation defines the few deviations: USB PID value, and the
+/// handling of OT reset signal.
+pub struct ServoMicroFlavor;
+
+impl ServoMicroFlavor {
+    const PID_SERVO_MICRO: u16 = 0x501A;
+}
+
+impl Flavor for ServoMicroFlavor {
+    fn gpio_pin(inner: &Rc<Inner>, pinname: &str) -> Result<Rc<dyn GpioPin>> {
+        if pinname == "IO_EXP_16" {
+            return Ok(Rc::new(ServoMicroResetPin::open(inner)?));
+        }
+        StandardFlavor::gpio_pin(inner, pinname)
+    }
+
+    fn spi_index(_inner: &Rc<Inner>, instance: &str) -> Result<(u8, u8)> {
+        bail!(TransportError::InvalidInstance(
+            TransportInterfaceType::Spi,
+            instance.to_string()
+        ))
+    }
+
+    fn get_default_usb_vid() -> u16 {
+        VID_GOOGLE
+    }
+
+    fn get_default_usb_pid() -> u16 {
+        Self::PID_SERVO_MICRO
+    }
+    fn perform_initial_fw_check() -> bool {
+        // The servo micro firmware and hyperdebug firmware are different, so do not check.
+        false
+    }
+}
+
+/// A register for the IO expander on the servo micro i2c bus
+#[derive(Copy, Clone)]
+#[repr(u8)]
+enum IoExpanderRegister {
+    /// Input values for Port 1 pins. `1'b` represents a logic high.
+    Input1 = 1,
+    /// Output values for Port 1 pins. `1'b` represents a logic high. Note that if `FloatPin1` bit
+    /// is set to high, then output pin value is not used.
+    Output1 = 3,
+    /// Specifies if the IO expander is floating or driving the pin. `1'b` represents the IO
+    /// expander producing a HighZ floating state on the pin, and `0'b` represents the IO expander
+    /// driving the value from the `Output1` register on the pin.
+    FloatPin1 = 7,
+}
+
+/// Handles the specialized IO expander logic to read and write to the OT reset pin
+pub struct ServoMicroResetPin {
+    inner: Rc<Inner>,
+}
+
+impl ServoMicroResetPin {
+    /// The titian reset pin mask for `IoExpanderRegister  `
```

Comment:
Nit: spelling, but we probably don't want to mention the name of the chip.  Maybe:
```
/// The target reset pin mask for `IoExpanderRegister`
```

Reply:
Done. Thanks!